src/Middleware/OutputCaching/src/CacheVaryByRules.cs

```diff
@@ -20,10 +20,15 @@ public sealed class CacheVaryByRules
     /// </summary>
     public IDictionary<string, string> VaryByCustom => _varyByCustom ??= new();
 
+    /// <summary>
+    /// Gets or sets the list of route values to vary by.
+    /// </summary>
+    public StringValues RouteValues { get; set; }
+
     /// <summary>
     /// Gets or sets the list of headers to vary by.
     /// </summary>
-    public StringValues Headers { get; set; }
+    public StringValues HeaderNames { get; set; }
```

Comment:
I do like that this name makes it more clear that header names rather than header values are in the property, but once again this name makes the implication stronger that `RouteValues` are actually values instead of `RouteValueNames` like they really are.

I think we should either consistently use the `Names` (`Keys` for query is sufficiently similar imo) suffix with these APIs or not. So either: `RouteValueNames`, `HeaderNames`, `QueryKeys`. Or: `RouteValues`, `Headers`, `Query` (with good doc comments).

Reply:
FYI, [HttpRequest.RouteValues](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httprequest.routevalues?view=aspnetcore-6.0) is a dictionary of names and values.

RouteValueNames seems like the best choice.

Reply:
Doing the changes to add Names/Keys everywhere then, I prefer that too.